lib/reek/context_builder.rb

```diff
@@ -133,20 +148,12 @@ def process_defs(exp)
     # we also record to what the method call is referring to
     # which we later use for smell detectors like FeatureEnvy.
     #
-    # :reek:TooManyStatements: { max_statements: 7 }
-    # :reek:FeatureEnvy
     def process_send(exp)
-      method_name = exp.method_name
-      if exp.visibility_modifier?
-        element.track_visibility(method_name, exp.arg_names)
-      elsif exp.attribute_writer?
-        exp.args.each do |arg|
-          append_new_context(Context::AttributeContext, arg, exp)
-        end
-      else
-        append_new_context(Context::SendContext, exp, method_name)
+      if current_context.module_context?
```

Comment:
Actually, it began as a case statement over the type of `current_context`. I now think I should have left it like that to make the type checking at least explicit for later improvement.